src/coordinate_systems/XYZT.jl

```diff
@@ -0,0 +1,174 @@
+"""
+
+    XYZT <: AbstractCoordinateSystem
+
+Cartesian coordinate system for four-vectors. Using this requires the implementation of the following interface functions:
+
+```julia
+LorentzVectorBase.x(::CustomFourVector)
+LorentzVectorBase.y(::CustomFourVector)
+LorentzVectorBase.z(::CustomFourVector)
+LorentzVectorBase.t(::CustomFourVector)
+```
+
+"""
+struct XYZT <: AbstractCoordinateSystem end
+coordinate_names(::XYZT) = (:x, :y, :z, :t)
+
+####
+# Interface functions
+####
+
+function t end
+
+function x end
+
+function y end
+
+function z end
+
+####
+# derived components
+####
+
+@inline function spatial_magnitude2(::XYZT, mom)
+  return x(mom)^2 + y(mom)^2 + z(mom)^2
+end
+
+@inline function spatial_magnitude(cs::XYZT, mom)
+  return sqrt(spatial_magnitude2(mom))
+end
+
+@inline function mass2(::XYZT, mom)
+  return t(mom)^2 - x(mom)^2 - y(mom)^2 - z(mom)^2
+end
+
+@inline function mass(::XYZT, mom)
+  m2 = mass2(mom)
+  if m2 < zero(m2)
+    # Think about including this waring, maybe optional with a global PRINT_WARINGS switch.
+    #@warn("The square of the invariant mass (m2=P*P) is negative. The value -sqrt(-m2) is returned.")
+    return -sqrt(-m2)
+  else
+    return sqrt(m2)
+  end
+end
+
+@inline function boost_beta(::XYZT, mom)
+  en = t(mom)
+  rho = spatial_magnitude(mom)
+  if !iszero(en)
+    rho / en
+  elseif iszero(rho)
+    return zero(rho)
+  else
+    throw(
+      ArgumentError(
+        "There is no beta for a four-vector with vanishing time/energy component"
+      ),
+    )
+  end
+end
+
+@inline function boost_gamma(::XYZT, mom)
+  beta = boost_beta(mom)
+  return inv(sqrt(one(beta) - beta^2))
+end
+
+########################
+# transverse coordinates
+########################
+@inline function pt2(::XYZT, mom)
+  return x(mom)^2 + y(mom)^2
+end
+
+@inline function pt(::XYZT, mom)
+  return sqrt(pt2(mom))
+end
+
+@inline function mt2(::XYZT, mom)
+  return t(mom)^2 - z(mom)^2
+end
+
+function mt(::XYZT, mom)
+  mT2 = mt2(mom)
+  if mT2 < zero(mT2)
+    # add optional waring: negative transverse mass -> -sqrt(-mT2) is returned.
+    -sqrt(-mT2)
+  else
+    sqrt(mT2)
+  end
+end
+
+@inline function rapidity(::XYZT, mom)
+  en = t(mom)
+  zcomp = z(mom)
+  return 0.5 * log((en + zcomp) / (en - zcomp))
+end
+
+function eta(::XYZT, mom)
+  cth = cos_theta(mom)
+
+  if cth^2 < one(cth)
+    return -0.5 * log((1 - cth) / (1 + cth))
+  end
+
+  z = z(mom)
+  if iszero(z)
+    return zero(z)
+  end
+
+  @warn "Pseudorapidity (η): transverse momentum is zero! return +/- 10e10"
+  if z > zero(z)
+    return 10e10
+  end
+
+  return 10e-10
```

Comment:
As I wrote below, I think it would make more sense to have the 1e10 as a general maximum (absolute) value that is returned. This should also be easy to implement as a simple `min(res, 1e10)` since `x/0.0` simply returns `Inf` which works with `min` as expected. You just need to take care of the correct sign.

In any case, currently instead of -10e10, 10e-10 is returned. Also I presume it should be 1e10 or 1e11 instead of 10e10.

Reply:
I guess this follows the implementation of [`ROOT::TVector3::PseudoRapidity`](https://root.cern/doc/v606/TVector3_8cxx_source.html#l00372), which is used in `ROOT::TLorentzVector`. Therefore, it's written in this form. 

However, I don't like this, because in Julia we have `Inf<:AbstractFloat` which is, in my opinion, way better than having an arbitrary constant to guard against `pt=0`. 

I would like to leave it like this and open an issue to discuss, how to handle these edge cases in general.